src/views/Nft/market/Profile/ConnectedProfile.tsx

```diff
@@ -32,13 +29,15 @@ const TabMenuWrapper = styled(Box)`
 `
 
 const ConnectedProfile = () => {
-  const { profile, isLoading: isProfileLoading } = useProfile()
-  const { achievements, achievementFetchStatus } = useAchievements()
   const { account } = useWeb3React()
-  const { userNftsInitializationState, nfts: userNfts } = useUserNfts()
-
-  useFetchAchievements()
-  useFetchUserNfts()
+  const { isLoading: isProfileLoading, profile } = useProfile()
+  const {
+    achievements,
+    isFetching: isAchievementsFetching,
+    refresh: refreshAchievement,
+  } = useAchievementsForAddress(account)
```

Comment:
At this moment,  both `ConnectedProfile` and `UnconnectedProfile` use this hook.

Could we level `useAchievementsForAddress` up to the parent ?

Reply:
If we put one level up and if address of that account is invalid we redundantly send achievement requests to node.

Reply:
I see. We could create a separate component for this line `<>{isConnectedProfile ? <ConnectedProfile /> : <UnconnectedProfile />}</>`. 

It would be a reasonable thing to do.